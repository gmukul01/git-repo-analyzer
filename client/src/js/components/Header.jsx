import React, { PureComponent } from "react";
import { Link } from "react-router-dom";

import { LOGOUT_URL } from "constants/urls";

export default class Header extends PureComponent {
  render() {
    const { userName } = this.props;
    return (
      <nav className="header">
        <Link to="/" className="logo">
          Git Repo Analyzer
        </Link>
        {userName && (
          <span>
            {userName} <a href={LOGOUT_URL}>Logout</a>
          </span>
        )}
      </nav>
    );
  }
}
