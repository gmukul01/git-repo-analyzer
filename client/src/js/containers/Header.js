import { connect } from "react-redux";

import Header from "components/Header";

const mapStateToProps = state => ({
  userName: state.user ? state.user.name : undefined
});

export default connect(
  mapStateToProps,
  null
)(Header);
