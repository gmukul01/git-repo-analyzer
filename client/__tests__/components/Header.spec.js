import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";

import Header from "components/Header";

describe("Header Component", () => {
  let wrapper;
  const initialProps = {
    userName: "dummyUserName"
  };
  it("should match snapshot when user is logged in", () => {
    const renderer = new ShallowRenderer(),
      result = renderer.render(<Header {...initialProps} />);
    expect(result).toMatchSnapshot();
  });

  it("should match snapshot when source is not present in props", () => {
    const props = {
        ...initialProps,
        userName: undefined
      },
      renderer = new ShallowRenderer(),
      result = renderer.render(<Header {...props} />);
    expect(result).toMatchSnapshot();
  });
});
