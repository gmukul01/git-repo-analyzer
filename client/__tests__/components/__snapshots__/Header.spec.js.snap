// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should match snapshot when source is not present in props 1`] = `
<nav
  className="header"
>
  <Link
    className="logo"
    replace={false}
    to="/"
  >
    Git Repo Analyzer
  </Link>
</nav>
`;

exports[`Header Component should match snapshot when user is logged in 1`] = `
<nav
  className="header"
>
  <Link
    className="logo"
    replace={false}
    to="/"
  >
    Git Repo Analyzer
  </Link>
  <React.Fragment>
    <section>
      <Link
        replace={false}
        to="/"
      >
        Home
      </Link>
      <Link
        replace={false}
        to="/search"
      >
        Search
      </Link>
      <a
        href="/auth/logout"
      >
        Logout
      </a>
    </section>
    <span>
      Hello 
      dummyUserName
    </span>
  </React.Fragment>
</nav>
`;
